### DISCUSSION TITLE
How payment gateway’s compatibility determined?

### DISCUSSION THREAD

**Discussion by jorgelucasjs** [woocommerce/woocommerce#51204] [WooCommerce, Blocks, and Site Editing]
How is a woocommerce payment gateway’s compatibility determined?

**Comment by moon0326** [comment]
Hi @jorgelucasjs 

> How is a woocommerce payment gateway’s compatibility determined?

Are you trying to create a custom payment gateway? If so, https://developer.woocommerce.com/docs/woocommerce-payment-gateway-api/ should help. The doc provides an example of a custom payment gateway and how to register it in WooCommerce.

**Comment by senadir** [comment]
Hey @jorgelucasjs if you're wondering how a payment gateway is shown as compatible in Checkout block editor page, then we do a simple compare. We check if a PHP registered gateway has a corresponding JS registered gateway.
